src/services/socketIoServices.ts

```diff
@@ -0,0 +1,169 @@
+'use client'
+
+import { io } from 'socket.io-client'
+import { useEffect, useState } from 'react'
+
+export type ListenerRes = {
+	messageServer: string
+	roomIdServer: string
+	userNameServer: string
+}
+
+export interface Event extends ListenerRes {
+	timeStamp: string
+}
+
+const socket = io()
+
+export function socketRoomEmitter(
+	emitterName: string,
+	message: string,
+	roomId: string,
+	userName: string,
+) {
+	socket.emit(emitterName, message, roomId, userName)
+}
+
+export function useSocketListener(listenerName: string) {
+	const [listenerRes, setListenerRes] = useState<ListenerRes>({
+		messageServer: '',
+		roomIdServer: '',
+		userNameServer: '',
+	})
+	useEffect(() => {
+		function onListenerRes(
+			messageServer: string,
+			roomIdServer: string,
+			userNameServer: string,
+		) {
+			setListenerRes({ messageServer, roomIdServer, userNameServer })
+		}
+		socket.on(listenerName, onListenerRes)
+
+		return () => {
+			socket.off(listenerName, onListenerRes)
+		}
+	}, [listenerName])
+
+	return listenerRes
+}
+
+// Only use this listener for the SocketIoInfo component to display
+// events for testing purposes
+export function useListenerEvents(joinListener: string, dataListener: string) {
```

Comment:
tip(docs): Instead of writing `//` to comment functions, prefer using [JSDoc](https://jsdoc.app/about-getting-started). Where you write `/**` and then when hovering over the function name it shows that description in the tooltips.

Reply:
We use that at work. I can do that.

Reply:
I'll make that change after we merge this branch. Like I said, I need help figuring out how to make use of multiple branches. Or not having multiple branches.